### ISSUE TITLE
undefined is not an object

### ISSUE THREAD

**Issue by erikparr** [google-ar/WebARonARKit#52]
Hi - Very nice project here. I'm exploring your examples and I keep getting the following error on plane finder example. I am on WebARonARKit, ios 11.2.5 three.js .89, and three.ar.js 1.7
![img_0038](https://user-images.githubusercontent.com/896122/35988015-a871fb00-0cfd-11e8-8412-12adad3e1e7c.PNG)

**Comment by judax** [comment]
Thank you for reaching out and pointing this problem out. Are you able to consistently reproduce the issue?

**Comment by erikparr** [comment]
Yes - consistent every time showPlanes is enabled for THREE.ARDebug

**Comment by ITJesse** [comment]
I can confirm this error.
It happens every new suface be find after reload the page in the WKWebView.

I have a workaround, hope it helps.
https://github.com/ITJesse/WebARonARKit/commit/833a90d7679a0375a699700fe330f3ab59e8b966

**Comment by lincolnfrog** [comment]
Thanks, @ITJesse - I left a comment on your fix. This is a deeper problem as basically we are getting the events out-of-order, which can result in all kinds of badness like recreating deleted planes if we get an update after a delete. The correct fix is to handle the events on the native side where they come in the correct order and then send a batch add/update/remove event to JS along with each frame update so it can't come in out-of-order on the JS side.

**Comment by lincolnfrog** [comment]
@ITJesse I left another comment - would like to get a band-aid in there but I think the best fix is to change onPlaneUpdated_, where the crash happens, to just ignore the update if the plane doesn't exist. That will prevent us from leaking planes if the update/delete come in reverse order. Do you want to make that fix?

**Comment by ITJesse** [comment]
I will try it later.
Thanks.

**Comment by ITJesse** [comment]
@lincolnfrog I have done some researchs these days. And I found that it maybe caused by a "dummy restart" of ARSession. [See here](https://github.com/google-ar/WebARonARKit/blob/ac9909d2a95e5d77e2ab7d4e46d53c7362ad82a1/WebARonARKit/WebARonARKit/ViewController.m#L693).

All of the planes has found is still left in the native side. This will cause the data to be out of sync between native and WKWebkit side after a web page reload happened (maybe out-of-memory, refresh, and so on).

So when we found a plane again, the native will fire a onPlaneUpdate event not a onPlaneAdded event.
